File: pt_app/models/events.py
import time
from sqlalchemy import (Column, Integer, String,
                        Boolean, BigInteger, ForeignKey,
                        DateTime, Float, UUID, func, ARRAY)
from sqlalchemy.dialects.postgresql import JSONB
from sqlalchemy.orm import relationship

from models.utils import BaseModel


class EventConsumer(BaseModel):
    __tablename__ = 'event_consumers'

    id = Column(Integer, primary_key=True, autoincrement=True)
    obj_consumer_id = Column(Integer, ForeignKey('obj_consumers.id'), nullable=True)
    source = Column(String, nullable=True)
    description = Column(String, nullable=True)
    is_approved = Column(Boolean, nullable=True)
    is_closed = Column(Boolean, nullable=True)
    probability = Column(Float, nullable=True)
    days_of_work = Column(Float, nullable=True)
    created = Column(DateTime(timezone=True), nullable=True)
    closed = Column(DateTime(timezone=True), nullable=True)


class EventType(BaseModel):
    __tablename__ = 'event_types'

    id = Column(Integer, primary_key=True, autoincrement=True)
    event_name = Column(String, nullable=False)


class EventCounter(BaseModel):
    __tablename__ = 'event_counters'

    id = Column(Integer, primary_key=True, autoincrement=True)
    obj_consumer_id = Column(Integer, ForeignKey('obj_consumers.id'), nullable=True)
    event_consumer_id = Column(Integer, ForeignKey('event_consumers.id'), nullable=True)

    contour = Column(String, nullable=True)
    counter_mark = Column(String, nullable=True)
    counter_number = Column(Integer, nullable=True)
    created = Column(DateTime(timezone=True), nullable=True)
    gcal_in_system = Column(Float, nullable=True)
    gcal_out_system = Column(Float, nullable=True)
    subset = Column(Float, nullable=True)
    leak = Column(Float, nullable=True)
    supply_temp = Column(Float, nullable=True)
    return_temp = Column(Float, nullable=True)
    work_hours_counter = Column(Float, nullable=True)
    heat_thermal_energy = Column(Float, nullable=True)
    errors = Column(String, nullable=True)
    errors_desc = Column(String, nullable=True)

File: go_app/pkg/models/events.go
package models

import "time"

type EventConsumer struct {
	ID            uint64         `gorm:"primaryKey" json:"id"`
	ObjConsumerId uint64         `json:"obj_consumer_id"`
	Source        string         `json:"source"` // additional source model
	Description   string         `json:"description"`
	IsApproved    bool           `gorm:"default:true;" json:"is_approved"`
	IsClosed      bool           `gorm:"default:true;" json:"is_closed"`
	Probability   float32        `gorm:"default:100.0;" json:"probability"`
	DaysOfWork    float64        `json:"days_of_work"`
	EventsCounter []EventCounter `json:"events_counter"`
	Created       time.Time      `gorm:"autoUpdateTime" json:"created"`
	Closed        time.Time      `json:"closed"`
}

type EventType struct {
	ID        uint64 `gorm:"primaryKey" json:"id"`
	EventName string `gorm:"unique" json:"event_name"`
}

type EventCounter struct {
	ID                uint64    `gorm:"primaryKey" json:"id"`
	ObjConsumerId     uint64    `json:"obj_consumer_id"`
	EventConsumerId   uint64    `json:"event_consumer_id"`
	Contour           string    `json:"contour"`
	CounterMark       string    `json:"counter_mark"`
	CounterNumber     int64     `json:"counter_number"`
	Created           time.Time `json:"created"`
	GcalInSystem      float64   `json:"gcal_in_system"`
	GcalOutSystem     float64   `json:"gcal_out_system"`
	Subset            float64   `json:"subset"`
	Leak              float64   `json:"leak"`
	SupplyTemp        float64   `json:"supply_temp"`
	ReturnTemp        float64   `json:"return_temp"`
	WorkHoursCounter  float64   `json:"work_hours_counter"`
	HeatThermalEnergy float64   `json:"heat_thermal_energy"`
	Errors            string    `json:"errors"`
	ErrorsDesc        string    `json:"errors_desc"`
}
